### ISSUE TITLE
Meanings of metric and scoring_functions in "AutoSklearnClassifier"?

### ISSUE THREAD

**Issue by jmren168** [automl/auto-sklearn#1188]
When I use a predefined metric (say metric_A), should I set metric = A or scoring_functions = [A]?

Followings are three scenarios, which one can be used to re-define the metric of the results of a classifier in AutoSklearn?
Thanks a lot.

scenario 1
automl = autosklearn.classification.AutoSklearnClassifier(
    time_left_for_this_task=600,
    per_run_time_limit=60,
    tmp_folder='/tmp/autosklearn_resampling_example',
    output_folder='/tmp/autosklearn_resampling_example_out',
    disable_evaluator_output=False,
    seed=1,
    metric=metric_A,
    scoring_functions=[metric_A],
}

scenario 2
automl = autosklearn.classification.AutoSklearnClassifier(
    time_left_for_this_task=600,
    per_run_time_limit=60,
    tmp_folder='/tmp/autosklearn_resampling_example',
    output_folder='/tmp/autosklearn_resampling_example_out',
    disable_evaluator_output=False,
    seed=1,
    metric=metric_A,
}

scenario 3
automl = autosklearn.classification.AutoSklearnClassifier(
    time_left_for_this_task=600,
    per_run_time_limit=60,
    tmp_folder='/tmp/autosklearn_resampling_example',
    output_folder='/tmp/autosklearn_resampling_example_out',
    disable_evaluator_output=False,
    seed=1,
    scoring_functions=[metric_A],
}

**Comment by eddiebergman** [comment]
Hi @jmren168, I imagine you have looked at the documentation [here](https://automl.github.io/auto-sklearn/master/api.html#autosklearn.classification.AutoSklearnClassifier) for the parameters. The `metric` is what is used for evaluation and producing the final ensemble where as the `scoring_functions` are for tracking additional scores as the pipelines are evaluated.

Scenario 1 and Scenario 2 would both accomplish your task if I understand correctly.

I do however agree that the parameter names are confusing and this should be documented better or provide clearer parameter names.

**Comment by jmren168** [comment]
Hi @eddiebergman, thank you for the quick reply and clearer explanation. 

In scenario 2, when I set **metric_A as "greater _is_better = True"** and **metric_B as "great_is_better = False** using cross-validation (where metric_B returns the minus result of metric_A), I got result 1 and result 2, respectively

What confused me is that If I understand **Best validation score** correctly, these two settings should obtain similar results.
But **Best validation score** seems to record the best **minus** result under cross-validation?

**Result 1** (metric_A as "greater _is_better = True")
auto-sklearn results:
  Dataset name: d
  Metric: metric_A
  **Best validation score: -0.002202**
  Number of target algorithm runs: 105
  Number of successful target algorithm runs: 101

**Result 2** (metric_B as "great_is_better = False)
auto-sklearn results:
  Dataset name: d
  Metric: metric_B
  **Best validation score: -1.782574**
  Number of target algorithm runs: 322
  Number of successful target algorithm runs: 310

metric_A and metric_B are defined as follows
def metric_A(sol, pred):
    tpr=...
    fpr=...
    return 2*tpr - fpr ### greater is better

def metrc_B(sol, pred):
   tpr=..
   fpr=...
   **return -(2*tpr-fpr)** ### smaller is better

**Comment by mfeurer** [comment]
Could you please briefly state where you read "Best validation score" from?
